### PR TITLE
Show detailed information in shops and inventory

### PR DESCRIPTION
**Pull request by ezsh**
Adds a setting to toggle displaying detailed item information inline or as tooltips:
setting

And that looks as follows.
Tooltips (default):
tooltip

Inlined:
details

### CODE REVIEW

**Comment by ezsh** [comment]
I'm not really sure about this, as the change uncovers game mechanics, but this bit of information greatly helps onboard Salty Mermaid.

**Comment by ezsh** [comment]
Maybe we better add this as a planned time until next port in the map passage?

**Comment by JJM152** [comment]
> Maybe we better add this as a planned time until next port in the map passage?

Not a bad idea, but I don't mind either way.

In terms of the shop restock, people have been asking for this for ages, so I don't mind putting it in as an optional setting. But yes, generally I don't want to reveal too many game mechanics. This one is fine though as a setting.

**Comment by ezsh** [comment]
Don't like to fill settings with micro-choices. Let me think about other possibilities for a while.

**Comment by JJM152** [comment]
There was a request also (that I agreed to) for putting known effects under items in the shop ui.

Something like this:
image

I think that's do-able, with a little nice formatting (indent a few spaces, make it a smaller text). Maybe widen the store UI a little bit to boot.

Both of these things could be covered under one option "More information in stores".

**Comment by ezsh** [comment]
> "More information in stores".

Agree. I'll take a look.

**Comment by ezsh** [comment]
> There was a request also (that I agreed to) for putting known effects under items in the shop ui.

Done, see the first post for details.

**Comment by ezsh** [comment]
@Jamie-Martin, please have a look. Does it suit your request?

File: src/000-SCRIPT_OBJ/Helpers.js [hunk 1]
```diff
@@ -251,18 +251,19 @@ App.PR = new function() {
         this.pEffectMeter = function(effect, item)
         {
             var output = "";
+            var effectStr = effect.replace(/ /g, '&nbsp;');
```
Comment (ezsh) [inline src/000-SCRIPT_OBJ/Helpers.js]: Not elegant, but otherwise rendering can split description right here, and it looks terrible in tooltips.

**Comment by Jamie-Martin** [comment]
Yes, that display of player-known info about items is great. I'm not entirely sure I like the restocking indicator, though, even if I hate having to recheck the galley. My idea was that new stock is in would be the indicator before getting to the store page, but then you'd have to store the last visit day to each store. Especially if restocking becomes based on docking or even random events, like quests or a future fishing event... you have to stay fishing while being buggered over a side rail, if you want to make it perverted. With the sailing ability, you might delay or speed up a restocking. 

As in, restocking days might change, but you get in the presence of the NPC and they tell you they have new stuff without having to click each day on the store. (Same way the Journal needs some indicator that a quest is ready to turn in.)

**Comment by ezsh** [comment]
Thank you for the response! I like the idea to add an indicator for new stock. However, information about days left is valuable on itself as it helps to plan activities during that time period.

**Comment by Jamie-Martin** [comment]
Definitely, as long as that was information you want to expose. It was kind of tense to be running low on food in the early game and not know when there's going to be new stew (as I try to avoid the gruel and grog). But days until restock does work.

**Comment by ezsh** [comment]
> as long as that was information you want to expose.

Hm.... Maybe expose it only when the shop owner mood is high enough?

File: src/200-WIDGETS/storeInventory.twee [hunk 2]
```diff
@@ -5,13 +5,27 @@
 <<set _ST = _SE.OpenStore(setup.player, _NPC);>>\
 <<set _CI = _ST.GetCommonInventory();>>\
 <<set _RI = _ST.GetRareInventory(); >>\
+<<set _DUR = _ST.DaysUntilRestocking() >>\
+<<if _DUR > 0>>\
+	<<set _restockingText to "Restocking in " + _DUR + " days">>\
+<<else>>\
+	<<set _restockingText to "Fresh supplies arrived this morning!">>\
+<</if>>\
+<<if SugarCube.settings.inlineDetailsInShop>>\
```
Comment (ezsh) [inline src/200-WIDGETS/storeInventory.twee]: When I tried to put <<if>> inside of the <td>, it escaped quotes and broke the table. Hence two variables are always set and used below unconditionally. I would appreciate a help to simplify this.

**Comment by ezsh** [comment]
> That makes a lot of sense, if that's information the shop owner knows.

Made the changes. Restocking info is shown when mood of the owner is higher than "Happy" (80).

**Comment by ezsh** [comment]
PR updated: tooltips (unconditional) added to the inventory listing as well. Tooltips frame colour changed, as the white one was creating a visual mess in long inventory lists superposing on top of the white letters. Text alignment in the tooltips changed to left instead of center; this looks better when showing ranks for gear items.

Screenshot of the updated tooltip:
inventory-tooltip

**Comment by Jamie-Martin** [comment]
This looks good, but I'd love a toggle to have the inventory also show the known stats beyond the tooltip. Ideally in the future, we could sort inventory by stat change.

**Comment by ezsh** [comment]
Inventory list is usually longer than that in shops, and inline details make it look awful. IMO, we need yet another option in the settings dialog if we enable that. Wouldn't it be to many options for the simple table output. 

> Ideally in the future, we could sort inventory by stat change.

There are too many stats to show columns for all of them in the inventory/shop tables, perhaps you meant another option?

**Comment by Jamie-Martin** [comment]
> 
> 
> Inventory list is usually longer than that in shops, and inline details make it look awful. IMO, we need yet another option in the settings dialog if we enable that. Wouldn't it be to many options for the simple table output.

I wouldn't know if it's too much for table output, but it's definitely a situation later in the game that you might be looking through your inventory to modify certain stats. I remember face prettier being one that took forever to raise for the bimbo 2 quest. Or you might have it sorted by energy for extensive whoring efforts. Tooltips help, but view most of your food/.potion stats at a glance OR sort by the wanted stat would help. Or even, perhaps, just some consistent default sort options for them, because there's no muscle memory for where the item you want is in your list.

> 
> > Ideally in the future, we could sort inventory by stat change.
> 
> There are too many stats to show columns for all of them in the inventory/shop tables, perhaps you meant another option?

One crazy idea was to search based on the inline display. Another was a drop down box at the top. I was thinking you'd need an ascending/descending checkbox or button as well, but some of the stats are separate, like femininity, masculinity, nutritition, and some have up and down options, like toxicity, or body parts.

**Comment by ezsh** [comment]
I gave up and unified inventory and shop looks.

**Pull request merged**